### ISSUE TITLE
can't start lizzie

### ISSUE THREAD

**Issue by krv76** [featurecat/lizzie#121]
Double click on lizzie.jar (open with java.exe) gives nothing.
java.exe -jar lizzie.jar 
gives the following output:_
java.io.IOException: Problem reading font data.
	at java.awt.Font.createFont0(Unknown Source)
	at java.awt.Font.createFont(Unknown Source)
	at wagner.stephanie.lizzie.gui.LizzieFrame.<clinit>(LizzieFrame.java:85)
	at wagner.stephanie.lizzie.Lizzie.main(Lizzie.java:43)
I'm using windows 7, jre-8u172-windows-x64 , lizzie 0.4
Plese help, thank you!

**Comment by Aerospace91** [comment]
I have personally downloaded both the Win64 CPU and GPU releases and get that error via command prompt. With and without running in admin through command prompt

**Comment by featurecat** [comment]
This may hold an answer as a replacement for line 85 of LizzieFrame: https://stackoverflow.com/questions/16621750/using-custom-fonts-java-io-ioexception-error-reading-font-data

I will update the release soon with this possible fix

**Comment by featurecat** [comment]
@Aerospace91, what happened when you tried to install the fonts? did you get an error message?

**Comment by featurecat** [comment]
Please try this Lizzie jar and let me know if it works. Thanks. https://drive.google.com/open?id=1v2OJQ1w3OK7cb8sqJPuFIo6qhG84aZV_

**Comment by featurecat** [comment]
Oops did not mean to close until receiving feedback!

**Comment by featurecat** [comment]
@mibitzi the error message should have changed slightly. Could you post it?

**Comment by Aerospace91** [comment]
@featurecat 
New Error Message:
Creating config file
Leelaz process ended.
java.io.IOException: Problem reading font data.
        at java.awt.Font.createFont0(Unknown Source)
        at java.awt.Font.createFont(Unknown Source)
        at wagner.stephanie.lizzie.gui.LizzieFrame.<clinit>(LizzieFrame.java:86)
        at wagner.stephanie.lizzie.Lizzie.main(Lizzie.java:43)

Also when I installed the fonts, the install themselves went fine, error message was still the same as the original poster.

**Comment by levavft** [comment]
same problem, windows 10 x64, lizzie gpu, just installed newest java.
Leelaz process ended.
java.io.IOException: Problem reading font data.
        at java.awt.Font.createFont0(Unknown Source)
        at java.awt.Font.createFont(Unknown Source)
        at wagner.stephanie.lizzie.gui.LizzieFrame.<clinit>(LizzieFrame.java:87)
        at wagner.stephanie.lizzie.Lizzie.main(Lizzie.java:43)

doesn't work with double click either.

**Comment by Dessicated** [comment]
Same things for me
double click doesn't work
Leelaz process ended.
java.io.IOException: Problem reading font data.
        at java.awt.Font.createFont0(Unknown Source)
        at java.awt.Font.createFont(Unknown Source)
        at wagner.stephanie.lizzie.gui.LizzieFrame.<clinit>(LizzieFrame.java:86>

        at wagner.stephanie.lizzie.Lizzie.main(Lizzie.java:43)

**Comment by featurecat** [comment]
[Does this one work?](https://drive.google.com/open?id=1v2OJQ1w3OK7cb8sqJPuFIo6qhG84aZV_) I disabled font loading entirely.

**Comment by krv76** [comment]
I've changed permissions (allow writeing to averyone) on the windows temp folder and now java -jar lizzie.jar works fine. Doubleclick on lizzie.jar or prompting java.exe lizzie.jar still doesn't work, but I can do batch for this.

**Comment by featurecat** [comment]
does the jar I just linked work? or is this the old jar?

**Comment by Dessicated** [comment]
The problem persists for me, even with the writing permission and the new jar.
I have only "Leela process ended" in error report when i use the command prompt.

**Comment by Koseng** [comment]
Yeah I also further investigated and figured out it is the missing network file. We all were so excited, great new Lizzie Version, Go Go Go Go. Download as last time and just run.

I also overlooked the hint to the necessary download of the network file. The hint for 0.4 should be above the picture and really highlighted.

**Comment by featurecat** [comment]
that's a relief. I see. Next version will probably have automatic network downloading anyway so hopefully it won't have that issue again! Glad you're excited for lizzie :))

**Comment by Koseng** [comment]
The readme states download and rename the network to "network". I'm quite sure a lot of users will just rename the xxxxxx.gz file to "network". And I'm also quite sure 50% of the windows users do not know how to unpack a .gz file.
-> You should really ship a network file in 0.4 or a 0.41 until there is the automatic download. Otherwise there will be a lot of frustrated users.

**Comment by DonMathi** [comment]
I have more or less tried everything, my version is still not running.
I have downloaded the latest version of the network, renamed it to network, placed it in the right folder. I have used my own leelaz_opencl_tuning file also. Still not running. I also tried to run it in cmd with administrator rights. Still no luck.

Leelaz process ended.
java.io.IOException: Problem reading font data.
        at java.awt.Font.createFont0(Unknown Source)
        at java.awt.Font.createFont(Unknown Source)
        at wagner.stephanie.lizzie.gui.LizzieFrame.<clinit>(LizzieFrame.java:87)
        at wagner.stephanie.lizzie.Lizzie.main(Lizzie.java:43)

**Comment by Dessicated** [comment]
Lizzie 0.4 finally work for me :smile: .
Originally i couldn't launch it. I had install the network version before going to post here.
But when i try to use the network from Lizzie 0.3 and going to see Lizzie working well i have understand that it is now became a problem from the network.

For extract it properly from Windows 7 i had to open the gz format in the interface of the software 7zip and rename it directly from them, no other way or software that i have try had work for me.

Maybe it is more easy from the prompt command ?

Thank for your follow-up.

**Comment by Koseng** [comment]
@DonMathi copy the file "network" from lizzie 0.3 to your lizzie 0.4 folder and try with that. It should work. If that works for you, you did not correctly extract the latest version of the network from the .gz file. As @Dessicated stated on windows that can be done with the free software 7-zip.

**Comment by featurecat** [comment]
DonMathi, according to the exception you posted, you didn't use the latest .jar that I posted in this thread. [this one (click this link)](https://drive.google.com/open?id=1v2OJQ1w3OK7cb8sqJPuFIo6qhG84aZV_)

@Koseng, @Dessicated, @mibitzi: did you have to use the .jar I linked in this thread, or the one that's available through Releases?

**Comment by olbrichj** [comment]
The provided jar above fixes the fontloading exception for me. Starting in cmd it throws an awt exception. "IllegalArgumentException: board Length may not be less than 18, but was 0.

**Comment by featurecat** [comment]
@olbrichj @Aerospace91 @mibitzi what is your operating system and JDK version? (run "java -version" in commandline to find out). I'm not sure what the original cause of the font reading exception was. Maybe I'll have to settle for using the system default font every time...

**Comment by mibitzi** [comment]
Can confirm that the original jar works with the correct network file, it was just the exception which was a bit confusing.

Windows 10 64bit

```
>java -version
java version "1.8.0_131"
Java(TM) SE Runtime Environment (build 1.8.0_131-b11)
Java HotSpot(TM) 64-Bit Server VM (build 25.131-b11, mixed mode)
```

**Comment by kingvagabond** [comment]
The master branch checkout and built with maven, run under JDK 10.0.1 in windows 10, got font data error. The provided download jar run directly got "Leelaz process ended" then done. Neither is working now.

**Comment by featurecat** [comment]
@kingvagabond can you please verify that you also downloaded the network, unzipped it, named it "network", and placed it in the same directory as the jar file you're running?

**Comment by kingvagabond** [comment]
@featurecat Thanks for the reply. I changed to JDK 8 and downloaded the network file, this time I got message "This version of Leela Zero is incompatible with Lizzie. Please follow the instructions in the readme." I think this is okay now - I need get the Leela Zero rebuild with replaced file.

**Comment by kingvagabond** [comment]
@featurecat Replace leelaz.exe with binary provided here, for leelaz-64-CPU.exe, renamed it to leelaz.exe and run lizzie, or rename leelaz-64-GPU.exe to leelaz.exe, the CLI throws error as:

Exception in thread "AWT-EventQueue-0" java.lang.IllegalArgumentException: boardLength may not be less than 18, but was 0
        at wagner.stephanie.lizzie.gui.BoardRenderer.calculatePixelMargins(BoardRenderer.java:486)
        at wagner.stephanie.lizzie.gui.BoardRenderer.calculatePixelMargins(BoardRenderer.java:712)
        at wagner.stephanie.lizzie.gui.BoardRenderer.convertScreenToCoordinates(BoardRenderer.java:758)
        at wagner.stephanie.lizzie.gui.LizzieFrame.onMouseMoved(LizzieFrame.java:555)
        at wagner.stephanie.lizzie.gui.Input.mouseMoved(Input.java:56)
        at java.awt.Component.processMouseMotionEvent(Component.java:6578)
        at java.awt.Component.processEvent(Component.java:6302)
        at java.awt.Container.processEvent(Container.java:2238)
        at java.awt.Window.processEvent(Window.java:2025)
        at java.awt.Component.dispatchEventImpl(Component.java:4889)
        at java.awt.Container.dispatchEventImpl(Container.java:2296)
        at java.awt.Window.dispatchEventImpl(Window.java:2746)
        at java.awt.Component.dispatchEvent(Component.java:4711)
        at java.awt.EventQueue.dispatchEventImpl(EventQueue.java:760)
        at java.awt.EventQueue.access$500(EventQueue.java:97)
        at java.awt.EventQueue$3.run(EventQueue.java:709)
        at java.awt.EventQueue$3.run(EventQueue.java:703)
        at java.security.AccessController.doPrivileged(Native Method)
        at java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:80)
        at java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:90)
        at java.awt.EventQueue$4.run(EventQueue.java:733)

**Comment by kingvagabond** [comment]
@featurecat There is anothe error:

java.io.IOException: Cannot run program "./leelaz" (in directory "."): CreateProcess error=216, This version of %1 is not compatible with the version of Windows you're running. Check your computer's system information and then contact the software publisher
        at java.lang.ProcessBuilder.start(ProcessBuilder.java:1048)
        at wagner.stephanie.lizzie.analysis.Leelaz.<init>(Leelaz.java:98)
        at wagner.stephanie.lizzie.Lizzie.lambda$main$0(Lizzie.java:32)
        at java.lang.Thread.run(Thread.java:748)

**Comment by featurecat** [comment]
interesting... try replacing "./leelaz" with "leelaz.exe" in Leelaz.java?

**Comment by featurecat** [comment]
If that doesn't work, try compiling Leela Zero yourself according to the readme 

Building Leela Zero
First, you will need to have a special version of Leela Zero that continually outputs pondering information. You can get this from one of the Lizzie releases or build it yourself; just substitute leelaz-src/UCTSearch.cpp for the UCTSearch.cpp file found in the Leela Zero sources. Then copy the resulting leelaz to the root of the Lizzie directory tree.

**Comment by olbrichj** [comment]
java version 1.8.0_171-b11
Windows 8.1

**Comment by featurecat** [comment]
Update: seems like this whole font bug was, unexpectedly, caused by not having the correct network file.

**Comment by levavft** [comment]
Will you put an assert at the start of the code, to check if the network file is set up properly?

**Comment by hydrogenpi** [comment]
The hardware for LZ is 6 x 1080ti. But the weight file is not decided yet. The report (in Chinese) could be found here http://www.yikeweiqi.com/news/topline/44206/

**Comment by BHydden** [comment]
@mibitzi thank you so much! I couldn't figure out why I was having so much trouble, turns out I forgot to unzip network before changing its file name haha whoops working now

**Comment by Tim0306028** [comment]
@ featurecat
I need to report to you the problem is still there.
One of my computers is OK, but the other one is not.
I have already tried the latest jar you shared on April 25.
The network I used is sure a right one.
The error message is only "Leela process ended" if I use cmd.
I have tried version 0.5 GPU, 0.5 CPU, 0.4 GPU, and 0.4 CPU, but all the same.....
Hope to know the answer in the future.
Good job to you anyway :)
